src/components/UserPofile/UserProfile.tsx

```diff
@@ -0,0 +1,87 @@
+import React, { FormEvent, useEffect } from "react";
+import { useRouter } from "next/router";
+import { useAuthStore } from "@/store/auth";
+import { Button } from "../Controls/Button";
+import { InputMasked } from "../Input/InputMasked";
+import { Preloader } from "../Preloader/Preloader";
+import style from "./UserCabinet.module.css";
+import classNames from "classnames";
+import { useUpdateProfile } from "../../hooks/useProfileUpdate";
+import { useFileUpload } from "../../hooks/useFileUpload";
+import { usePhoneNumber } from "../../hooks/usePhoneNumber";
+
+export const UpdateProfile: React.FC = () => {
+  const { user } = useAuthStore();
+  const router = useRouter();
+
+  const { loading, successMessage, errorMessage, handleUpdateProfile } = useUpdateProfile(user);
+  const { phoneNumber, handlePhoneNumberChange } = usePhoneNumber(user?.phoneNumber || "");
+  const { fileInputRef, fileName, imageError, handleFileInputChange, handleUpload, getShortFileName } = useFileUpload();
+
+  const handleSubmit = async (event: FormEvent) => {
+    event.preventDefault();
+    await handleUpdateProfile(phoneNumber);
+  };
+
+  useEffect(() => {
+    if (!user) {
+      router.push("/403"); 
+    }
+  }, [user, router]);
+
+  return (
+    <div className={style.form_content_wrapper}>
+      {loading ? (
+        <Preloader />
+      ) : (
+        <form className={style.form} onSubmit={handleSubmit}>
+          <h3 className={style.title}>Update your profile here</h3>
+          <div className={style.form_content_wrapper}>
+            <InputMasked
+              id="phoneNumber"
+              value={phoneNumber}
+              onChange={handlePhoneNumberChange}
+              type="tel"
+              placeholder="Enter your phone number" 
+              errorMessage={undefined} 
+              name={""} 
+              onBlur={function (event: React.FocusEvent<HTMLInputElement>): void {
```

Comment:
You should make onBlur optional in InputMasked component, because now it would throw error each time user blur input